BUILDING.txt

```diff
@@ -63,6 +63,20 @@ Installing required packages for clean install of Ubuntu 14.04 LTS Desktop:
   $ sudo apt-get -y install maven
 * Native libraries
   $ sudo apt-get -y install build-essential autoconf automake libtool cmake zlib1g-dev pkg-config libssl-dev libsasl2-dev
+* GCC 9.3.0
+  $ sudo apt-get -y install software-properties-common
+  $ sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+  $ sudo apt-get update
+  $ sudo apt-get -y install g++-9 gcc-9
+  $ sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 --slave /usr/bin/g++ g++ /usr/bin/g++-9
+* CMake 3.19
+  $ mkdir -p /opt/cmake \
+    && curl -L https://cmake.org/files/v3.19/cmake-3.19.0.tar.gz > cmake-3.19.0.tar.gz \
+    && tar -zxvf cmake-3.19.0.tar.gz \
+    && cd cmake-3.19.0 \
+    && ./bootstrap \
+    && make -j$(nproc) \
+    && sudo make install
```

Comment:
We need sudo for creating directory under /opt and the /opt/cmake is not used. I think we don't need to apply Dockerfile practice in this instruction. The Protocol Buffers section below is not a good example. It seemed to be copied from Dockerfile and is assuming root privilege.
same for CentOS 8 part.

Reply:
Done.

Reply:
> The Protocol Buffers section below is not a good example. It seemed to be copied from Dockerfile and is assuming root privilege.

I've fixed this, please review - https://github.com/apache/hadoop/pull/2664.